backend/siarnaq/api/user/views.py

```diff
@@ -143,3 +145,31 @@ def avatar(self, request, pk=None):
             titan.upload_image(avatar, profile.get_avatar_path())
 
         return Response(None, status=status.HTTP_204_NO_CONTENT)
+
+
+class TeamByUserViewSet(
+    viewsets.GenericViewSet,
+):
+    """
+    A viewset for retrieving a user's associated teams.
+    """
+
+    permission_classes = (AllowAny,)
+    filter_backends = [filters.SearchFilter]
+
+    def get_queryset(self):
+        return Team.objects.all()
+
+    @action(
+        detail=False,
+        permission_classes=(IsAuthenticated,),
```

Comment:
Would `AllowAny` be enough here? I think non-logged-in people can also be allowed to see team memberships.